src/context/UserLetterContext.tsx

```diff
@@ -0,0 +1,149 @@
+import { createContext, PropsWithChildren } from "react";
+import { DatabaseLetter, LetterInterface, LetterType } from "../types";
+import { useStickyState } from "../utils/localstorage";
+import { Letters } from "../data/letters";
+import { useEffect, useState } from "react";
+import { LetterFormButton } from "../components/LetterForm";
+import { supabase } from "../lib/supabaseClient";
+import { SubmitLetterMetadata } from "../components/Home";
+
+interface UserLetterContextType {
+  letters: LetterInterface[];
+  fromName: string;
+  toName: string;
+  fromStamp: string;
+  content: string;
+  type: LetterType;
+  setFromName: (fromName: string) => void;
+  setToName: (toName: string) => void;
+  setFromStamp: (fromStamp: string) => void;
+  setContent: (content: string) => void;
+  setType: (setType: LetterType) => void;
+  onLetterSubmitted: () => void;
+}
+
+type PersistedUserLetterContextInfo = Pick<
+  UserLetterContextType,
+  "fromName" | "toName" | "fromStamp" | "content" | "type"
+>;
+
+const DefaultUserLetterContext = {
+  letters: [],
+  fromName: "",
+  toName: "the internet",
+  fromStamp: "",
+  content: "",
+  type: LetterType.IFrame,
+  setFromName: () => {},
+  setToName: () => {},
+  setFromStamp: () => {},
+  setContent: () => {},
+  setType: () => {},
+  onLetterSubmitted: () => {},
+};
+export const UserLetterContext = createContext<UserLetterContextType>(
+  DefaultUserLetterContext
+);
+
+const UserContextStorageId = "user-letter-context";
+
+export function UserLetterContextProvider({ children }: PropsWithChildren) {
+  const [userContext, setUserContext] =
+    useStickyState<PersistedUserLetterContextInfo>(
+      UserContextStorageId,
+      DefaultUserLetterContext
+    );
+
+  const { fromName, toName, fromStamp, content, type } = userContext;
+  const setFromName = (fromName: string) =>
```

Comment:
would it make sense to use a reducer here? no strong pref but I think it makes more sense esp if we plan on changing this more

Reply:
wait what does that mean

Reply:
jk i realized this was zustand and not react LMAO ignore